applications/DEMApplication/tests/test_DEM_2D_restitution.py

```diff
@@ -0,0 +1,177 @@
+import os
+import KratosMultiphysics
+from KratosMultiphysics import Logger
+Logger.GetDefaultOutput().SetSeverity(Logger.Severity.WARNING)
+from KratosMultiphysics.DEMApplication import *
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.DEMApplication.DEM_analysis_stage
+
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+this_working_dir_backup = os.getcwd()
+
+def GetFilePath(fileName):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
+
+def CreateAndRunStageInOneOpenMPThread(my_obj, model, parameters_file_name):
+    omp_utils = KratosMultiphysics.OpenMPUtils()
+    if "OMP_NUM_THREADS" in os.environ:
+        initial_number_of_threads = os.environ['OMP_NUM_THREADS']
+        omp_utils.SetNumThreads(1)
+
+    with open(parameters_file_name,'r') as parameter_file:
+        project_parameters = KratosMultiphysics.Parameters(parameter_file.read())
+
+    my_obj(model, project_parameters).Run()
+
+    if "OMP_NUM_THREADS" in os.environ:
+        omp_utils.SetNumThreads(int(initial_number_of_threads))
+
+class DEM2D_RestitutionTestSolution(KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage):
+
+    def Initialize(self):
+        super(DEM2D_RestitutionTestSolution, self).Initialize()
+        for node in self.spheres_model_part.Nodes:
+            self.initial_normal_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
+
+    @classmethod
+    def GetMainPath(self):
+        return os.path.join(os.path.dirname(os.path.realpath(__file__)), "DEM2D_restitution_tests_files")
+
+    def GetProblemNameWithPath(self):
+        return os.path.join(self.main_path, self.DEM_parameters["problem_name"].GetString())
+
+    @classmethod
+    def CheckRestitution(self, reference, restitution_coefficient, tolerance):
+        if not (reference < abs(restitution_coefficient*tolerance) and reference > abs(restitution_coefficient/tolerance)):
+            raise ValueError('Incorrect value for COEFFICIENT_OF_RESTITUTION: expected value was '+ str(reference) + ' but received ' + str(restitution_coefficient))
+
+    def Finalize(self):
+        tolerance = 1.0001
+        for node in self.spheres_model_part.Nodes:
+            final_vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y)
+            print("initial velocity:",self.initial_normal_vel)
+            print("final velocity:",final_vel)
+            restitution_coefficient = -final_vel / self.initial_normal_vel
```

Comment:
All these prints should use the KratosLogger. Otherwise we will see too many things on screen